pkg/api/context.go

```diff
@@ -86,3 +100,8 @@ func WithNamespaceDefaultIfNone(parent Context) Context {
 	}
 	return parent
 }
+
+// withTrace returns a copy of the parent in which the trace value is set
+func withTrace(parent Context, trace string) Context {
```

Comment:
Hrm - i don't think it has to be public since the new pattern is that the *only* way you can get a context (or should) is to get a traced one - i.e. it's the thing that starts the others.  Let me rethink it.

----- Original Message -----
> > @@ -86,3 +100,8 @@ func WithNamespaceDefaultIfNone(parent Context) Context
> > {
> >  	}
> >  	return parent
> >  }
> > +
> > +// withTrace returns a copy of the parent in which the trace value is set
> > +func withTrace(parent Context, trace string) Context {
> 
> Meant to be public?
> 
> ... I cannot fathom why the google context package uses functions like this
> and not functions that take parent as a receiver.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3826/files#r24847302